motion/README.md

```diff
@@ -0,0 +1,122 @@
+to print them into the terminal, or to a machine-readable format, namely JSON, using `Boost.JSON`. 
+
+## Running examples
+We provide all three examples in a [tutorial directory in MOTION](https://github.com/encryptogroup/MOTION/blob/master/src/examples/tutorial), 
+so they are not included here.
+To run our examples, you'll need inputs to the circuit. We've written an input
+generation script for `mult3` and other tutorial examples that allows to quickly test the examples. 
+It will tell you what the expected result is.
+```
+$ python ~/MOTION/src/examples/tutorial/geninput.py -e <example>
+Expected result: 621
+```
+
+Executables are located in the `MOTION/<build>/bin/` directory. 
+In general, at least two parties must run each example.
+You will find more details regarding the settings enabled in particular 
+tutorial examples in the [tutorial README](https://github.com/encryptogroup/MOTION/blob/master/src/examples/tutorial/README.md). 
+You can do it in one line as here, or in two separate terminals.
+```
+$ cd ~/MOTION/<build>/bin
+$ my_parties="0,<ip0>,<port0> 1,<ip1>,<port1>"
+$ ./<example> --my-id 0 --parties $my_parties & 
+$ ./<example> --my-id 1 --parties $my_parties
+...
+Result: 621
+```
+To run this locally, replace `<ip0>` and `<ip1>` with 127.0.0.1.
+
+For settings with more than two parties, the configurations may get quite complicated and inconvenient to use from the command line.
+For this case, it is possible to store one or more configurations in files and input them via the command line.
+The arguments in config files have the format `my-id=0`, one per line.
+Command line inputs are preferred over the config file in case of a conflict.
+
+For more info on command line arguments, enter `<example>_main --help`.
+
+## Modifying examples
+The fastest way is to copy/modify the `example_template` in `MOTION/src/examples`. 
+Change the target name `example_template` to `<your_example>` in `CMakeLists.txt` and rename the source files correspondingly. 
+If you only want to play around with the framework without real inputs and evaluate some circuit, 
+you would only need to put your function in `example_template.cpp:EvaluateProtocol()`, 
+similarly to the other provided examples.
+To pass real inputs to your circuit, we recommend to use `Boost.ProgramOptions` in the `<example>_main.cpp`,
+which allows to either pass inputs directly or pass a filename that contains the inputs. 
+
+To build your example, add it to `MOTION/src/examples/CMakeLists.txt` and re-run `make` in the `MOTION/<build>` directory.
+
+For an overview of the classes and their connections in MOTION, check out the [online doxygen documentation](https://motion-documentation.github.io).
+Also, we are actively working on a developer guide that will contain more implementation guidelines and details about MOTION.
+It will soon be available [here](https://motion-documentation.github.io/motiondevguide.pdf).
```

Comment:
```suggestion

```

I suggest removing this paragraph and re-adding it when the developer guide  becomes available